aips/higher-order-functions.md

```diff
@@ -0,0 +1,132 @@
+---
+aip: TBD
+title: Higher-Order Inline Functions for Collections
+author: wrwg
+discussions-to: https://github.com/aptos-foundation/AIPs/issues/33
+Status: Draft
+last-call-end-date: TBD
+type: Standard (Framework)
+created: 2023/1/9
+updated: 2023/1/9
+---
+
+## Summary
+
+Recently, the concept of *inline functions* has been added to Aptos Move.
+Those functions are expanded at compile time and do not have an equivalent
+in the Move bytecode. This allows them to implement a feature which is
+currently not available for regular Move functions: taking functions,
+given as lambda expressions, as parameters. Given this, we can
+define popular higher-order functions like for_each, filter, map, and fold
+collection types in Move. In this AIP, we suggest a set of conventions
+to add those functions to the data structure in the Move and Aptos frameworks.
```

Comment:
"to the data structure" sounds more foundational -- like all data structures
I think
"to existing data structures in"